airflow-core/src/airflow/dag_processing/collection.py

```diff
@@ -196,7 +196,7 @@ def _serialize_dag_capturing_errors(
             min_update_interval=settings.MIN_SERIALIZED_DAG_UPDATE_INTERVAL,
             session=session,
         )
-        if dag_was_updated:
+        if dag_was_updated and "FabAuthManager" in conf.get("core", "auth_manager"):
             _sync_dag_perms(dag, session=session)
         else:
             # Check and update DagCode
```

Comment:
or like this (can't suggest the update_source_code line, but you get the idea):

```suggestion
        if not dag_was_updated:
            DagCode.update_source_code(dag.dag_id, dag.fileloc)
            return []
        
        if "FabAuthManager" in conf.get("core", "auth_manager"):
            _sync_dag_perms(dag, session=session)
```